ChurchToolsApi.py

```diff
@@ -529,3 +529,67 @@ def get_event_agenda(self, event_id: int):
             return response_data
         else:
             logging.warning("Something went wrong fetiching events: {}".format(response.status_code))
+
+    def file_download(self, filename: str, domain_type, domain_identifier, path_for_download='./Downloads'):       
+        """
+        Retrieves file from ChurchTools for specific filename, domain_type and domain_identifier from churchtools
+        domain_type: Currently supported are 'avatar', 'groupimage', 'logo', 'attatchments', 'html_template', 'service', 'song_arrangement', 'importtable', 'person', 'familyavatar', 'wiki_.?'.
+        domainId = Id of Arrangement
+        These information can be requested for example by calling get_songs()
+        """
+        StateOK=False
+        CHECK_FOLDER = os.path.isdir(path_for_download)
+
+        # If folder doesn't exist, then create it.
+        if not CHECK_FOLDER:
+            os.makedirs(path_for_download)
+            print("created folder : ", path_for_download)
+
+        url = '{}/api/files/{}/{}'.format(self.domain, domain_type, domain_identifier)
+
+        response = self.session.get(url=url)
+
+        if response.status_code == 200:
+            response_content = json.loads(response.content)
+            arrangement_files = response_content['data'].copy()
+            logging.debug("SongArrangement-Files load successful {}".format(response_content))
+            fileUrl=None
+
+            for file in arrangement_files:
+                filenameoriginal = str(file['name'])
+                fileUrl= str(file['fileUrl'])
+                if filenameoriginal==filename:
+                    break;
+            if fileUrl==None:
+                logging.warning("File {} does not exist".format(filename))
+            else:
+                logging.debug("Found File: {}".format(filename))
+                # Build path OS independent
+                path_file = os.sep.join([path_for_download, filename])
+                StateOK = self.file_download_from_url(fileUrl,path_file)
+
+            return StateOK
+        else:
+            logging.warning("Something went wrong fetching SongArrangement-Files: {}".format(response.status_code))
+ 
+    def file_download_from_url(self, file_url: str, target_path: str):       
+        """
+        Retrieves file from ChurchTools for specific file_url from churchtools
+        This function is used by file_download(...)
+        Pay Attention: this file-url consists of a specific / random filename which was created by churchtools
+        Example -> file_url=https://lgv-oe.church.tools/?q=public/filedownload&id=631&filename=738db42141baec592aa2f523169af772fd02c1d21f5acaaf0601678962d06a00
```

Comment:
just an idea - but how about using the churchtools sample server for common references and try to use it for tests?